neuraxle/steps/logic.py

```diff
@@ -0,0 +1,134 @@
+
+        return filtered
+
+
+class If(NonFittableMixin, Pipeline):
+    def __init__(self, condition_step: BaseStep, then_step: BaseStep, else_step: BaseStep):
+        NonFittableMixin.__init__(self)
+        Pipeline.__init__(self, [
+            ('condition_step', condition_step),
+            ('then_step', then_step),
+            ('else_step', else_step)
+        ])
+
+    def _transform_data_container(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        if self['condition_step'].transform(data_container.data_inputs):
+            data_inputs = self['then_step'].transform(data_container.data_inputs)
+        else:
+            data_inputs = self['else_step'].transform(data_container.data_inputs)
+
+        data_container.set_data_inputs(data_inputs)
+
+        return data_container
+
+
+class GreaterOrEqual(NonFittableMixin, BaseStep):
+    def __init__(self, then, space: HyperparameterDistribution = None):
+        if space is None:
+            space = FixedHyperparameter(then)
+        BaseStep.__init__(
+            self,
+            hyperparams=HyperparameterSamples({'then': then}),
+            hyperparams_space=HyperparameterSpace({'then': space})
+        )
+        NonFittableMixin.__init__(self)
+
+    def transform(self, data_inputs):
+        return data_inputs >= self.hyperparams['then']
+
+
+class LessOrEqual(NonFittableMixin, BaseStep):
+    def __init__(self, then, space: HyperparameterDistribution = None):
+        if space is None:
+            space = FixedHyperparameter(then)
+        BaseStep.__init__(
+            self,
+            hyperparams=HyperparameterSamples({'then': then}),
+            hyperparams_space=HyperparameterSpace({'then': space})
+        )
+        NonFittableMixin.__init__(self)
+
+    def transform(self, data_inputs):
+        return data_inputs <= self.hyperparams['then']
+
+
+class Greater(NonFittableMixin, BaseStep):
+    def __init__(self, then, space: HyperparameterDistribution = None):
+        if space is None:
+            space = FixedHyperparameter(then)
+
+        BaseStep.__init__(
+            self,
+            hyperparams=HyperparameterSamples({'then': then}),
+            hyperparams_space=HyperparameterSpace({'then': space})
+        )
+        NonFittableMixin.__init__(self)
+
+    def transform(self, data_inputs):
+        return data_inputs > self.hyperparams['then']
+
+
+class Less(NonFittableMixin, BaseStep):
+    def __init__(self, then, space: HyperparameterDistribution = None):
+        if space is None:
+            space = FixedHyperparameter(then)
+        BaseStep.__init__(
+            self,
+            hyperparams=HyperparameterSamples({'then': then}),
+            hyperparams_space=HyperparameterSpace({'then': space})
+        )
+        NonFittableMixin.__init__(self)
+
+    def transform(self, data_inputs):
+        return data_inputs < self.hyperparams['then']
```

Comment:
This method's types and dimensions should absolutely be documented before being merged. What type and shape is data inputs?

Reply:
importnat.